xilem_core/src/message.rs

```diff
@@ -1,74 +1,99 @@
-// Copyright 2022 the Xilem Authors
+// Copyright 2024 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-use std::any::Any;
+//! Message routing and type erasure primitives.
 
-#[macro_export]
-macro_rules! message {
-    ($($bounds:tt)*) => {
-        pub struct Message {
-            pub id_path: xilem_core::IdPath,
-            pub body: Box<dyn std::any::Any + $($bounds)*>,
-        }
+use core::{any::Any, fmt::Debug};
+use std::ops::Deref;
 
-        impl Message {
-            pub fn new(id_path: xilem_core::IdPath, event: impl std::any::Any + $($bounds)*) -> Message {
-                Message {
-                    id_path,
-                    body: Box::new(event),
-                }
-            }
-        }
-    };
-}
+use alloc::boxed::Box;
 
-/// A result wrapper type for event handlers.
+/// The possible outcomes from a [`View::message`]
+///
+/// [`View::message`]: crate::View::message
 #[derive(Default)]
-pub enum MessageResult<A> {
-    /// The event handler was invoked and returned an action.
+pub enum MessageResult<Action> {
+    /// An action for a parent message handler to use
     ///
-    /// Use this return type if your widgets should respond to events by passing
-    /// a value up the tree, rather than changing their internal state.
-    Action(A),
-    /// The event handler received a change request that requests a rebuild.
-    ///
-    /// Note: A rebuild will always occur if there was a state change. This return
-    /// type can be used to indicate that a full rebuild is necessary even if the
-    /// state remained the same. It is expected that this type won't be used very
-    /// often.
-    #[allow(unused)]
+    /// This allows for sub-sections of your app to use an elm-like architecture
```

Comment:
How useful is this really?

Reply:
I'm not sure yet about it exactly, but I guess it (also) allows to modularize views, possibly in external libraries. Maybe we don't need it, when we have found a satisfying solution for local/modularized state.